### ISSUE TITLE
PrismaClient is unable to be run in the browser

### ISSUE THREAD

**Issue by Raybrilliant** [prisma/prisma#20691]
### Bug description

when I use prisma by calling prisma client on the next js server component then sending the results to the props client component, an error appears. but if I don't use the client component the error doesn't appear

![Screen Shot 2023-08-15 at 06 53 37](https://github.com/prisma/prisma/assets/86271770/1474ccf6-48ee-4269-9535-6ed8e5cc37fd)

### How to reproduce

1. Create db.ts with 

```
import { PrismaClient } from '@prisma/client'

const globalForPrisma = globalThis as unknown as {
  prisma: PrismaClient | undefined
}

export const prisma = globalForPrisma.prisma ?? new PrismaClient()

if (process.env.NODE_ENV !== 'production') globalForPrisma.prisma = prisma
```

2. Create getActions.ts 
```
import { prisma } from "./Database"

export const GetList = async (e:string = "")=>{
    const res = await prisma.movies.findMany()
    return res
}
```

3. Call on Server Component 
```
export default async function Home() {
  const getMovies = await GetList()
```

4. Throw return value to Client Component
```
{getMovies.map((item:any)=>(
        <PosterCard key={item.id} slug={item.link} title={item.title} image={item.poster} caption={item.year}/>
      ))}
```

5. See error

### Expected behavior

displays the results from getMovies.map on the PosterCard client component

![Screen Shot 2023-08-15 at 08 06 34](https://github.com/prisma/prisma/assets/86271770/60f99277-d090-4dee-bf7d-416681c09f0c)

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

model Movies {
  slug       String   @id
  title      String
  year       String
  poster     String
  film       String?
  tmdb_id    String
  watch      Int?     @default(0)
  like       Int?     @default(0)
  created_at DateTime @default(now())
  update_at  DateTime @updatedAt
}
```

```ts
import { prisma } from "./Database"

export const GetList = async (e:string = "")=>{
    const res = await prisma.movies.findMany()
    return res
}
```

### Environment & setup

- OS: MacOS 12
- Database: PlanetScale
- Node.js version: 18.17.0
- Next JS version : 13.4.16

### Prisma Version

```
prisma                  : 5.1.1
@prisma/client          : 5.1.1
Current platform        : darwin
Query Engine (Node-API) : libquery-engine 6a3747c37ff169c90047725a05a6ef02e32ac97e (at node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Schema Engine           : schema-engine-cli 6a3747c37ff169c90047725a05a6ef02e32ac97e (at node_modules/@prisma/engines/schema-engine-darwin)
Schema Wasm             : @prisma/prisma-schema-wasm 5.1.1-1.6a3747c37ff169c90047725a05a6ef02e32ac97e
Default Engines Hash    : 6a3747c37ff169c90047725a05a6ef02e32ac97e
Studio                  : 0.492.0

```

**Comment by masterbater** [comment]
Sadly prisma doesnt support edge without using their prisma data

**Comment by Jolg42** [comment]
@masterbater What do you mean by edge? It looks unrelated here, there is nothing mentioning edge in this issue.

**Comment by Jolg42** [comment]
@Raybrilliant This error happens locally, right? We got a few similar reports already, we're looking at improving this.

**Comment by masterbater** [comment]
> @masterbater What do you mean by edge? It looks unrelated here, there is nothing mentioning edge in this issue.
Sorry I dont know if there is an issue with prisma but I got this error when deploying cloudflare worker, so I assume this error is specific to edge environment
![image](https://github.com/prisma/prisma/assets/26339988/96808583-31a4-44cc-96a9-1dfb62bbfa36)

**Comment by Raybrilliant** [comment]
@Jolg42 yes this error happen locally, i just deploy in my localhost computer environment and not hosting it yet

**Comment by Jolg42** [comment]
@masterbater This is a different issue for you, and indeed you would need to use Prisma's Data Proxy or Accelerate for this to work, because you are using Prisma Client Edge build here.
If you want to discuss more about this, need help, open a discussion or a new issue, I'll mark the messages as off-topic to keep this issue "clean".

**Comment by Jolg42** [comment]
@Raybrilliant Could you provide a minimal reproduction as a git repo or so? It would help in getting a reproduction and improving this.

**Comment by Raybrilliant** [comment]
@Jolg42 sure this all my code, https://github.com/Raybrilliant/prisma-error.git

**Comment by hihhhello** [comment]
Not sure if it might be related but I run into the same error while using Prisma in the Next.js 13 API route. In the Nextauth route in particular

**Comment by maxswjeon** [comment]
A very ugly fix, but it works if this is a problem that block you from developing an application.

Change `db.ts` to the content below.

```
import { PrismaClient } from "@prisma/client";

const globalForPrisma = globalThis as unknown as {
  prisma: PrismaClient | undefined;
};

export const prisma: PrismaClient =
  typeof window === "undefined"
    ? globalForPrisma.prisma ?? new PrismaClient()
    : (undefined as unknown as PrismaClient);

if (process.env.NODE_ENV !== "production") globalForPrisma.prisma = prisma;
```

**Comment by janpio** [comment]
In Prisma 5.2 we implemented a similar solution, so @maxswjeon's workaround should not be necessary any more. 
Please let us know if this fixes your problem.

(You will still get an error message if you actually try to _use_ Prisma Client in those (for now) unsupported environments of course. But it will now also try to give more information and a better error message.)

**Comment by Jolg42** [comment]
I could reproduce and confirm it was fixed by upgrading to `5.2.0`
